submariner/pluggable-network-plugin-and-ovn.md

```diff
@@ -0,0 +1,300 @@
+## Add support for OVN with pluggable network-plugin support
+
+[submariner#778](https://github.com/submariner-io/submariner/issues/778)
+
+## Summary
+
+Today Submariner supports working with existing network-plugins which are based
+on the kube-proxy with iptables backend. Any network-plugin outside of this
+case won't work, OVN-Kubernetes is an example.
+
+This is a proposal to support pluggable network drivers, starting with OVN-Kubernetes.
+
+## Proposal
+
+Re-architect Submariner route-agent in a way such that different network-plugin implementations
+can be supported. Those drivers would be notified of different events like:
+
+* `Initialization` event
+    Called once during startup, to let the plugin configure any necessary settings on the
+    system once the route-agent is ready
+
+* `Stop` event
+    Called once during stop to let the network plugin cleanup anything necessary during
+    at stop if at all.
+
+* `Uninstall` event
+    Called once if the route agent is being removed from a node, being Submariner un-installed,
+    or the route-agent driver being switched, a full cleanup will be necessary in this case.
+
+* `Leader`/`Non-leader` events
+    Some parts of the drivers could need a single leader handling specific synchronization
+    work with the network plugin. Those events will be called for drivers that require
+    this type of behavior.
+
```

Comment:
Yes, probably it would make sense as a 2nd iteration. Specially since not all drivers would need something like that. 

I'm ok with documenting it for the long term and start by the simpler approach.

Reply:
I will add those comments.

Reply:
Ok, I refectored it, after thinking more about it I believe it makes more sense to spawn a separate deployment from the start just for this.